app/models/creator.rb

```diff
@@ -1,2 +1,67 @@
 class Creator < ActiveRecord::Base
+  validates_presence_of :name, :uid, :provider
+  validates_uniqueness_of :uid
+  has_many :categories
+  has_many :users, through: :categories
+  has_many :videos
+  has_many :tweets
+
+  def self.find_or_create(params)
+    creator = self.find_by(uid: params["uid"], provider: params["provider"])
+    if !creator.nil?
+      return creator
+    else
+      Creator.transaction do
```

Comment:
Cool! Never heard of this `transaction` rails magic before. Yay for learning new things.